### PR TITLE
[Feature][Connector-V2] Jdbc connector support SAP HANA Source & SAP HANA Sink.

### PR DESCRIPTION
**Pull request by FlechazoW**
## Purpose of this pull request

[Connector-V2][JDBC-connector] support Jdbc dm

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by liugddx** [comment]
Is it necessary to have init-datasource as a separate file?I think the paperwork is getting a little messy,How about writing it straight into code?

**Comment by FlechazoW** [comment]
> Is it necessary to have init-datasource as a separate file?I think the paperwork is getting a little messy,How about writing it straight into code?

Nice thought.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+            case SAP_HANA_TIMESTAMP:
+            case SAP_HANA_LONGDATE:
+            case SAP_HANA_SECONDDATE:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+            case SAP_HANA_DECIMAL:
+                return new DecimalType(precision, scale);
+            case SAP_HANA_REAL:
+            case SAP_HANA_SMALLDECIMAL:
+                return BasicType.FLOAT_TYPE;
+            case SAP_HANA_BIGINT:
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:84]: BasicType.LONG TYPE is best.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            case SAP_HANA_INTEGER:
+            case SAP_HANA_SMALLINT:
+            case SAP_HANA_TINYINT:
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:90]: BasicType.BYTE_TYPE is best.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            case SAP_HANA_SECONDDATE:
+                return LocalTimeType.LOCAL_DATE_TIME_TYPE;
+            case SAP_HANA_DECIMAL:
+                return new DecimalType(precision, scale);
+            case SAP_HANA_REAL:
+            case SAP_HANA_SMALLDECIMAL:
+                return BasicType.FLOAT_TYPE;
+            case SAP_HANA_BIGINT:
+            case SAP_HANA_INTEGER:
+            case SAP_HANA_SMALLINT:
```
Comment (liugddx) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:88]: BasicType.SHORT_TYPE is best.

**Review by Hisoka-X — DISMISSED** [review]
LGTM, Please take a look @ic4y

**Comment by Hisoka-X** [comment]
The code style should be fixed.

**Comment by FlechazoW** [comment]
> The code style should be fixed.

Done @Hisoka-X

**Comment by Hisoka-X** [comment]
@FlechazoW The Integration Test have some problems should be fixed.

File: seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcSapHanaIT.java [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+    private Connection jdbcConnection;
+
+    private GenericContainer<?> dbServer;
+
+    @Override
+    public void startUp() throws Exception {
+        dbServer = new GenericContainer<>(SAP_HANA_DOCKER_IMAGE)
+            .withNetwork(NETWORK)
+            .withNetworkAliases(SAP_HANA_CONTAINER_HOST)
+            .withLogConsumer(new Slf4jLogConsumer(log));
```
Comment (hailin0) [inline seatunnel-e2e/seatunnel-connector-v2-e2e/connector-jdbc-e2e/src/test/java/org/apache/seatunnel/connectors/seatunnel/jdbc/JdbcSapHanaIT.java]:
```suggestion
            .withLogConsumer(new Slf4jLogConsumer(DockerLoggerFactory.getLogger(SAP_HANA_DOCKER_IMAGE)));
```

https://github.com/apache/incubator-seatunnel/pull/3028

**Review by hailin0 — COMMENTED** [review]
LGTM

**Review by TyrantLucifer — DISMISSED** [review]
LGTM

**Comment by CalvinKirs** [comment]
trigger CI

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java (hunk 1 above)
Comment (ic4y) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:84]: `SAP_HANA_BIGINT` can be use `BasicType.LONG_TYPE`

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java (hunk 3 above)
Comment (ic4y) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:88]: `SAP_HANA_SMALLINT` ->  `BasicType.SHORT_TYPE`

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java (hunk 2 above)
Comment (ic4y) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:90]: `SAP_HANA_TINYINT` -> `BasicType.BYTE_TYPE`

**Review by ic4y — COMMENTED** [review]
Thanks for your contribution, please add e2e tests

**Review by Hisoka-X — DISMISSED** [review]
LGTM @ic4y PTAL.

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java (hunk 1 above)
Comment (hailin0) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/internal/dialect/saphana/SapHanaTypeMapper.java:84]: It seems unchanged?

**Review by TyrantLucifer — DISMISSED** [review]
LGTM, cc @ic4y

**Comment by ic4y** [comment]
Are `SAP_HANA_BIGINT`, `SAP_HANA_INTEGER`, `SAP_HANA_SMALLINT` all tested with `SHORT_TYPE`? doesn't look right

**Review by TyrantLucifer — APPROVED** [review]
LGTM, cc @hailin0 @liugddx @ic4y

**Review by liugddx — APPROVED** [review]

**Pull request merged**